writer/proto/resource.go

```diff
@@ -106,6 +108,26 @@ func GeneratedResourceMessage(r *parser.ParsedResource) (*builder.MessageBuilder
 			typ = builder.FieldTypeDouble()
 		case schema.Type_FLOAT:
 			typ = builder.FieldTypeFloat()
+		case schema.Type_OBJECT:
+			wantedType := fmt.Sprintf("%s/%s", s.Name, p.ObjectType)
+			_, ok := m.Messages[wantedType]
+			if(!ok) {
+				// Resource has not been generated yet.
+				c := CombineResourceMaps(s.ObjectByType, s.ResourceByType)
```

Comment:
this feels like an code smell to me - we have two distinct maps that contain very similar data, and are referred to in similar ways (via a unique name for that type).

I understand that we have a difference between objects and resources - but maybe we can consolidate into a single "ObjectByType" map and have a flag that lets you know if it's a "resource" or not?

Also technically there's nothing in aepc that would prevent us from resources with no standard methods, unaccessible from the API except as sub-objects. But that might in turn make it harder to disambiguate between user-intended resources for things like documentation. So maybe "every object is a resource" is harder to justify.

Reply:
Yeah, those are all great points. I was thinking we'd want to keep them strictly separate, but this is cleaner. Changes made!